docs/topics/db/transactions.txt

```diff
@@ -252,6 +252,126 @@ by Django or by third-party libraries. Thus, this is best used in situations
 where you want to run your own transaction-controlling middleware or do
 something really strange.
 
+Performing actions after commit
+===============================
+
+.. versionadded:: 1.9
+
+Sometimes you need to fire off an action related to the current database
+transaction, but only if the transaction successfully commits. Examples might
+include a `Celery`_ task, an email notification, or a cache invalidation.
+
+.. _Celery: http://www.celeryproject.org/
+
+Django provides the :func:`on_commit` function to register code
+that should be executed after a transaction is successfully committed:
+
+.. function:: on_commit(func, using=None)
+
+Pass any function (that takes no arguments) to :func:`on_commit`::
+
+    from django.db import transaction
+
+    def do_something():
+        pass  # send a mail, invalidate a cache, fire off a Celery task, etc.
+
+    transaction.on_commit(do_something)
+
+You can also wrap your thing up in a lambda::
+
+    transaction.on_commit(lambda: some_celery_task.delay('arg1'))
+
+The function you pass in will be called immediately after a hypothetical
+database write made at the same point in your code is successfully committed.
+
+If you register a callback while there is no transaction active, it will be
+executed immediately.
+
+If that hypothetical database write is instead rolled back (typically when an
+unhandled exception is raised in an :func:`atomic` block), your function will
+be discarded and never called.
+
+Order of execution
+------------------
+
+On-commit functions for a given transaction are executed in the order they were
+registered.
+
+
+Exception handling
+------------------
+
+If one on-commit hook within a given transaction raises an uncaught exception,
+no later-registered hooks in that same transaction will run. (This is, of
+course, the same behavior as if you'd executed the hooks sequentially yourself
+without :func:`on_commit`.)
+
+
+Timing of execution
+-------------------
+
+Your hook functions are executed *after* a successful commit, so if they fail,
+it will not cause the transaction to roll back. They are executed conditionally
+upon the success of the transaction, but they are not *part* of the
+transaction. For the intended use cases (mail notifications, Celery tasks,
+etc), this should be fine. If it's not (if your follow-up action is so critical
+that its failure should mean the failure of the transaction itself), then you
+don't want to use the :func:`on_commit` hook.  (Instead, you may want
+`two-phase commit`_.)
```

Comment:
I think it would be better if we had a Django example for "two-phase commit" instead of linking to the wikipedia article. Is it too complicated?